sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/DiskEncryptionSet.java

```diff
@@ -0,0 +1,215 @@
+        interface Blank extends DefinitionWithRegion<WithGroup> { }
+
+        /** The stage of a disk encryption set definition allowing to specify the resource group. */
+        interface WithGroup extends GroupableResource.DefinitionStages.WithGroup<WithEncryptionType> { }
+
+        /**
+         * The stage of a disk encryption set definition allowing to set the disk encryption set type.
+         */
+        interface WithEncryptionType {
+            /**
+             * Set the disk encryption set type.
+             * @param type the type of key used to encrypt the data of the disk
+             * @return the next stage of the definition
+             */
+            WithKeyVault withEncryptionType(DiskEncryptionSetType type);
+        }
+
+        /**
+         * The stage of a disk encryption set definition allowing to associate with an Azure key vault.
+         */
+        interface WithKeyVault {
+            /**
+             * Associates with the disk encryption set an Azure key vault by its resource ID.
+             * @param keyVaultId resource ID of the Azure key vault
+             * @return the next stage of the definition
+             */
+            WithKeyVaultKey withExistingKeyVault(String keyVaultId);
+        }
+
+        /**
+         * The stage of a disk encryption set definition allowing to associate with an Azure key vault key.
+         */
+        interface WithKeyVaultKey {
+            /**
+             * Associate with the disk encryption set an Azure key vault key by its ID.
+             * @param keyId ID of the Azure key vault key
+             * @return the next stage of the definition
+             */
+            WithSystemAssignedManagedServiceIdentity withExistingKey(String keyId);
+        }
+
+        /**
+         * The stage of a disk encryption set definition allowing to enable System Assigned Managed Service
+         * Identity.
+         */
+        interface WithSystemAssignedManagedServiceIdentity {
+            /**
+             * Specifies that System Assigned Managed Service Identity needs to be enabled in the disk
+             * encryption set.
+             *
+             * @return the next stage of the definition
+             */
+            WithSystemAssignedIdentityBasedAccessOrCreate withSystemAssignedManagedServiceIdentity();
+        }
+
+        /**
+         * The stage of the System Assigned Managed Service Identity enabled disk encryption set allowing to set
+         * access role for the key vault.
+         */
+        interface WithSystemAssignedIdentityBasedAccessOrCreate extends WithCreate {
+            /**
+             * Specifies that disk encryption set's system assigned identity should have the given RBAC based access
+             * (described by the role) on the current Azure key vault that's associated with it.
+             * Only works for key vaults that use the 'Azure role-based access control' permission model.
```

Comment:
use `<p>` to break the line, for summary and detail?

Reply:
ok

Reply:
added